### PR TITLE
Fix env_vars in QiskitPattern

### LINKED ISSUE #1283: Review environment variables functionality
### Steps to reproduce the problem

When we `upload` a `QiskitPattern` we allow to pass environment variables: `env_vars`. But we are not sending them to the gateway.

In the same way these `env_vars` should be managed by the `Job` and being encrypted when we add them.

### What is the expected behavior?

Store `env_vars` encrypted when we upload a `QiskitPattern`. Manage those `env_vars` in the Job when we create it.

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
fix #1283

### CODE REVIEW

File: client/quantum_serverless/core/job.py [hunk 1]
```diff
@@ -490,6 +491,7 @@ def upload(self, program: QiskitPattern):
                             "entrypoint": program.entrypoint,
                             "arguments": json.dumps({}),
                             "dependencies": json.dumps(program.dependencies or []),
+                            "env_vars": json.dumps(program.env_vars or {}),
```
Comment (Tansito) [inline client/quantum_serverless/core/job.py:494]: One question here: `env_vars` should be set with the `QiskitPattern` or in the `Job` like arguments? @psschwei @IceKhan13

**Comment by Tansito** [comment]
Regardless my question, Aki. Can we update some of the tests to check that we are storing correctly the `env_vars`?

File: client/quantum_serverless/core/job.py (hunk 1 above)
Comment (akihikokuroda) [reply client/quantum_serverless/core/job.py:494]: Here is my 2 cents.  For a configuration value, its default is in the environment variable with the pattern and the argument of the Job can overwrite it at the execution time.  So I think that the pattern is better have the env_vars.

**Comment by akihikokuroda** [comment]
The test is added.

**Review by Tansito — APPROVED** [review]
LGTM, thanks Aki!

**Pull request merged**